docs/env/ubuntu/24.04.md

```diff
@@ -0,0 +1,82 @@
+# Ubuntu 24.04 Setup Instructions
+
+## Verify Hardware Support
+
+Most modern GPUs support OpenCL. For integrated graphics devices (iGPUs), use `lscpu` to get the processor SKU. Detailed information for Intel SKUs is available from [ark.intel.com](ark.intel.com). Detailed information for AMD processors is available from [AMD's product page](https://www.amd.com/en/products/specifications/processors).
+
+## Build Dependencies
+
+OCL Headers:
+
+```bash
+sudo apt install opencl-c-headers opencl-clhpp-headers
+```
+
+The OpenCL API has its own set of header files; the above command installs both C and C++ headers files. The C header can be found in `<CL/cl.h>`; the C++ header is in `<CL/cl.hpp>`.
```

Comment:
Nitpick: I think the C++ header is now "opencl.hpp", not "cl.hpp".

```suggestion
The OpenCL API has its own set of header files; the above command installs both C and C++ headers files. The C header can be found in `<CL/cl.h>`; the C++ header is in `<CL/opencl.hpp>`.
```

Reply:
Ah, good point. Fixed!